blueprints/addon/index.js

```diff
@@ -142,8 +142,16 @@ module.exports = {
     let appFiles = this.lookupBlueprint(this.appBlueprintName).files();
     let addonFilesPath = this.filesPath(this.options);
     let addonFiles = walkSync(addonFilesPath);
+    let files = appFiles.concat(addonFiles);
 
-    return uniq(appFiles.concat(addonFiles));
+    // There will be two ember-cli-build.js present when app and addon
+    // files are concatenated - one at tests/dummy and project root.
+    let i = files.indexOf('ember-cli-build.js');
```

Comment:
I think I would tweak the assignment above:

```js
let files = [].concat(
  appFiles.filter(f => f !== 'ember-cli-build.js'),
  addonFiles,
);
```

Instead of splicing manually...